crates/wdk-build/src/lib.rs

```diff
@@ -922,18 +952,24 @@ impl Config {
     ///
     /// The contents contain `#include`'ed headers based off the [`ApiSubset`]
     /// and [`Config`], as well as any additional definitions required for the
-    /// headers to be processed successfully
+    /// headers to be processed successfully.
+    ///
+    /// # Errors
+    /// [`ConfigError`] - if the headers for a [`ApiSubset`] could not be
+    /// determined, or if formatting the header contents fails
     pub fn bindgen_header_contents(
         &self,
         api_subsets: impl IntoIterator<Item = ApiSubset>,
-    ) -> String {
+    ) -> Result<String, ConfigError> {
         api_subsets
             .into_iter()
-            .flat_map(|api_subset| {
-                self.headers(api_subset)
-                    .map(|header| format!("#include \"{header}\"\n"))
+            .try_fold(String::new(), |mut acc, api_subset| {
+                for header in self.headers(api_subset)? {
+                    writeln!(acc, "#include \"{header}\"")
+                        .map_err(ConfigError::BindgenHeaderContents)?;
```

Comment:
`ConfigError::BindgenHeaderContents` seems to only exist because of how this code is structured here (writeln). This string concatenation should be infallible, so this error shouldn't exist